pkg/controller/controller.go

```diff
@@ -169,6 +179,97 @@ func (c *Controller) getWorkflowUpdatesHandler() cache.ResourceEventHandler {
 	}
 }
 
+// This object is responsible for emitting metrics that show the current number of Flyte workflows, cut by project and domain.
+// It contains
+// It needs to be kicked off. The periodicity is not currently configurable because it seems unnecessary.
+type ResourceLevelMonitor struct {
+	Scope promutils.Scope
+
+	// Meta timer - this times each collection cycle
+	CollectorTimer promutils.StopWatch
+
+	// This is a labeled gauge
+	levels *prometheus.GaugeVec
+
+	// The thing that we want to current levels of
+	lister lister.FlyteWorkflowLister
+}
+
+func (r *ResourceLevelMonitor) countList(ctx context.Context, workflows []*v1alpha1.FlyteWorkflow) {
+	// Map of Projects to Domains to counts
+	counts := map[string]map[string]int{}
+
+	// Collect all workflow metrics
+	for i, wf := range workflows {
+		execID := wf.GetExecutionID()
+		if execID.WorkflowExecutionIdentifier == nil {
+			logger.Warningf(ctx, "Workflow does not have an execution identifier! [%v]", wf)
+			continue
+		}
+		fmt.Printf("in the loop %d: %s %s %s %s\n", i, wf.ExecutionID.Project, wf.ExecutionID.Domain, wf.GetID(), wf.ExecutionID.Name)
+		project := wf.ExecutionID.Project
+		domain := wf.ExecutionID.Domain
+		if _, ok := counts[project]; !ok {
+			counts[project] = map[string]int{}
+		}
+		counts[project][domain] += 1
+	}
+}
+
+func (r *ResourceLevelMonitor) collect(ctx context.Context) {
+	// Emit gauges at both the project/domain level, but also at the aggregate level
+	workflows, err := r.lister.List(labels.Everything())
+	if err != nil {
+		logger.Errorf(ctx, "Error listing workflows when attempting to collect data for gauges %s", err)
+	}
+
+	counts := r.countList(ctx, workflows)
+
+	// Emit labeled metrics, for each project/domain combination. This can be aggregated later with Prometheus queries.
+	metricKeys := []contextutils.Key{contextutils.ProjectKey, contextutils.DomainKey}
```

Comment:
Decided not to do phase for now @EngHabu - should we move this to stdlib?  It's not quite the same in the sense that I don't really care to emit the _unlabeled metric.

Reply:
unlabeled is an option.. when you create any of the labeled.New*, if you do not specify EmitUnlabeledOption, it won't...